pstore/api.go

```diff
@@ -34,11 +37,108 @@ type Record struct {
 	Timestamp time.Time
 }
 
+// Writer is the interface for writing to persistent store.
+// Implementations of Writer must be safe to use with multiple goroutines.
 type Writer interface {
+	// Write writes given collection of records to persistent storage
+	Write(records []Record) error
+}
```

Comment:
I started with s/Writer/PStoreWriter but remembered that when clients use this they will use pstore.Writer for my writer and io.Writer for the well known writer.  If I change the name, clients will have to use pstore.PStoreWriter, pstore.PStoreLimitedWriter etc. That may seem repetitive and verbose. In fact, the authors of "Effective Go" warn about this and encourage go developers to remember that the package name acts as part of the full name. You can read about this here https://golang.org/doc/effective_go.html#package-names in the 3rd paragraph.

Unless you feel very strongly about renaming, I'd prefer to leave as pstore.Writer